datasets/us/dod_chinese_milcorps/crawler.py

```diff
@@ -0,0 +1,26 @@
+from zavod import Context
+import csv
+
+def crawl(context: Context):
+    source_file = context.fetch_resource("source.csv", context.dataset.data.url)
+    with open(source_file, "r") as f:
+        reader = csv.DictReader(f)
+        for row in reader:
+            entity = context.make("Company")
+            name = row.pop("Name")
+            entity.id = context.make_id(name)
+            entity.add("name", name)
+            entity.add("alias", row.pop("Alias", None))
+            clean_name = row.pop("Clean Name", None)
+            if clean_name != "" and clean_name != name:
+                entity.add("alias", clean_name)
+            entity.add("notes", row.pop("Note", None))
+            parent_name = row.pop("Parent Name", None)
+            if parent_name != "" and parent_name != name:
+                parent = context.make("Company")
+                parent.id = context.make_id(parent_name)
+                parent.add("name", parent_name)
+                context.emit(parent)
+                entity.add("parent", parent)
+            context.emit(entity, target=True)
```

Comment:
before emitting we also usually add a [topic](https://www.opensanctions.org/reference/#type.topic) which helps users understand why an entity is listed.

```suggestion
            entity.add("topics", "debarment")
            context.emit(entity, target=True)
```

In this case, it seems it's not an immediate ban, except for some regulations stopping the DOD themselves from doing business with them. https://www.reuters.com/technology/pentagon-adds-companies-chinese-military-list-us-official-says-2024-01-31/ - @pudo does this mean we tag it `debarment` with a little paragraph in the description about the limits of the debarment?

is it ok to reference the reuters article or search a bit further for a gov source?

I think in the general case it's useful to indicate the impact of being listed in the dataset description - especially when it isn't straight up financial sanctions for terrorism or something, or straight up PEPs. But in a case like this or [this](https://www.opensanctions.org/datasets/si_conflicts/) where it's not an outright ban, it's even more important.

Reply:
Didn't find a gov source as such, but there was a source from some consultancy that I added.

Reply:
cool - let's go with the debarment topic, and could you add a sanction entity per company using `h.make_sanction()` setting the program to `Section 1260H of the William M. (“Mac”) Thornberry National Defense Authorization Act for Fiscal Year 2021 (Public Law 116-283)` and emit that as well?

We typically use the Sanction entity to describe the "sanction" against these companies even when it's just a debarment. When it's a true financial sanction, we use the `sanction` topic.